Automattic/wp-calypso
# Change endpoint to get domains and use emails api

#### Changes proposed in this Pull Request

This pull request is converting the emails-home class component to a React functional component, in order to use the new endpoint created in the `emails-api` in WPCOM, which at the present moment replicates the get domains for the domains-api.

The component needed to be functional in order to use the new endpoint as it is a react query component to get the information from the WPCOM backend.

#### Testing instructions

The emails-home is used in two places: Inbox screen and emails -> upgrade. therefore we need to assert that both screens are still working fine.

**Check that the Inbox screen loads fine in the following cases**
1. You have a site with domains and no emails, so it "redirects" you to emails comparison
2. You have a site with no domains, so it "redirects" you to buy new domains
3. You have a site with emails so you see a list of all your inboxes

**Check that Upgrade -> Emails screen loads correctly for the following cases**
1. You have WPCOM domains, so you see a list of those domains for domains with and without emails
2. You don't have WPCOM domains, so you are prompted to buy new domains


Related to {1200182182542585-as-1201750212847841}
